core/src/jmh/java/com/uber/m3/tally/AbstractReporterBenchmark.java

```diff
@@ -66,25 +78,43 @@
     private static final String HISTOGRAM_DURATION_NAME = "histogram_duration";
     private static final String HISTOGRAM_VALUE_NAME = "histogram_value";
 
-    private StatsReporter reporter;
+    private static final int TARGET_PARALLELISM = 4;
+
+    private Reporter reporter;
 
     @Benchmark
-    public void reportCounterBenchmark(IncrementalValueState value) {
+    public void reportCounterBenchmark(Counter value) {
         reporter.reportCounter(COUNTER_NAME, DEFAULT_TAGS, value.incrementAndGet());
     }
 
+    @Threads(TARGET_PARALLELISM)
     @Benchmark
-    public void reportGaugeBenchmark(IncrementalValueState value) {
-        reporter.reportCounter(COUNTER_NAME, DEFAULT_TAGS, value.incrementAndGet());
+    public void reportCounterParallelBenchmark(Counter value) {
+        if (isOpen()) {
+            reporter.reportCounter(COUNTER_NAME, DEFAULT_TAGS, value.incrementAndGet());
+        }
+    }
+
+    // This method is used to throttle the execution, returning
```

Comment:
nit: this is not a java doc and {@code ...} will be ignored, you could either change it to java doc or just remove @code blocks

Reply:
Yeah, just following the same structure for readability